packages/expo-cli/src/commands/upgrade.ts

```diff
@@ -227,6 +227,46 @@ async function upgradeAsync(requestedSdkVersion: string | null, options: Options
     }
   }
 
+  const platforms = exp.platforms || [];
```

Comment:
The fallback would also be a great reason for moving this to `.isPlatformEnabled` I think, since it may or may not be `undefined`.